plugins/zstack_stage.py

```
@@ -0,0 +1,276 @@
+            )
+
+            levels = generate_zlevels(
+                self.stage,
+                (zmin, zmax),
+                zstep
+            )
+
+            logging.info("Generated %d z-levels: %s", len(levels), levels)
+
+        except (ValueError, IndexError, ZeroDivisionError, KeyError) as e:
+            logging.exception("Failed to generate z-levels: %s", e)
```

Ruff (0.15.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)